### DISCUSSION TITLE
Unclear usage of --output-file with the main example

### DISCUSSION THREAD

**Discussion by MTRNord** [ggml-org/whisper.cpp#575] [Q&A]
Hi when using the main example with multiple files I somehow seem to not be able to properly use the --output-file flag. I am basically doing this:

```
files_already_done=($(find . -wholename "./output/*.txt" -type f | tr '\n' ' ' | sed 's/output/playlist_normalized/g'| sed 's/txt/wav/g'))
files=$(find . -wholename "./playlist_normalized/*.wav" -type f | tr '\n' ' ')
for already_done_file in "${files_already_done[@]}"
do
    files=$(echo "${files}" | sed "s@${already_done_file}@@g")
done
out_files=$(echo "${files}" | sed 's/playlist_normalized/output/g' | sed 's/.wav//g')

./main -m "models/ggml-${MODEL}.bin" -t "$THREADS" -l en -otxt -pc --file $files --output-file $out_files
```
The variables I generate are just an overengineered way to make sure I don't pass it files I already processed. Other than that, I just generate a space separated string of files.

However, apart from the first wav file, all wav files go into `playlist_normalized/<wav_name>.txt` instead of the desired `output/<video_id>.txt`.

I wonder is this because I incorrectly use the flag?

**Comment by mirozahorak** [comment]
-o simply expects just a basename used for all output formats and it adds the extensions according to selected output format f.e. .csv for -ocsv, .vtt for -ovtt etc.

**Reply by MTRNord** [reply]
Ah Hm. But how do I make it that it matches the input file name.

I have a list of input files with different names that all are supposed to be written with their respective output filename to the output folder. And I can't figure out how to do that properly :)
